processing/src/test/java/org/apache/druid/segment/join/JoinConditionAnalysisTest.java

```diff
@@ -60,6 +61,7 @@ public void test_forExpression_simple()
         ImmutableList.of(),
         exprsToStrings(analysis.getNonEquiConditions())
     );
+    Assert.assertThat(analysis.getRightKeyColumns(), CoreMatchers.is(ImmutableList.of("y")));
```

Comment:
Is this better than `Assert.assertEquals`?

Reply:
assertEquals fails because getRightKeyColumns doesn't return an ImmutableList

Reply:
How can that be? `CoreMatchers.is` is calling `actual.equals(expected)`, with extra steps.

Reply:
I definitely saw the test fail which is why I had to google how to fix this - and I found CoreMatchers.

hahaha I think I know why now -
`Assert.assertEquals` checks `expected.equals(actual)`
`CoreMatchers.is` checks `actual.equals(expected)`
That probably means the equals implementation for one (or both?) of those classes isn't correct according to the javadocs